pkg/upstream/http.go

```diff
@@ -100,17 +101,37 @@ func newReverseProxy(target *url.URL, upstream options.Upstream, errorHandler Pr
 	if upstream.FlushInterval != nil {
 		proxy.FlushInterval = *upstream.FlushInterval
 	} else {
-		proxy.FlushInterval = 1 * time.Second
+		proxy.FlushInterval = options.DefaultFlushInterval
 	}
 
+	var finalTimeout time.Duration
+	if upstream.Timeout != nil {
+		finalTimeout = *upstream.Timeout
+	} else {
+		finalTimeout = options.DefaultTimeout
+	}
+
+	var finalKeepAlive time.Duration
+	if upstream.Timeout != nil {
+		finalKeepAlive = *upstream.KeepAlive
+	} else {
+		finalKeepAlive = options.DefaultKeepAlive
+	}
+
+	defaultTransportClone := http.DefaultTransport.(*http.Transport).Clone()
+	defaultTransportClone.DialContext = (&net.Dialer{
+		Timeout:   finalTimeout,
+		KeepAlive: finalKeepAlive,
+	}).DialContext
+
 	// InsecureSkipVerify is a configurable option we allow
 	/* #nosec G402 */
 	if upstream.InsecureSkipTLSVerify {
-		proxy.Transport = &http.Transport{
```

Comment:
Hmmm, I wonder if this old code that made a new `http.Transport` with just `TLSClientConfig` configured, but the rest the struct defaults led to improper settings compared to using `http.DefaultTransport` that set sensible timeouts and limits